homeassistant/components/sensor/greeneye_monitor.py

```diff
@@ -127,7 +127,7 @@ def _on_new_monitor(self, *args):
             monitors.remove_listener(self._on_new_monitor)
 
     def _try_connect_to_monitor(self, monitors):
-        monitor = monitors.monitors.get(self._monitor_serial_number)
+        monitor = monitors.monitors.get(int(self._monitor_serial_number))
```

Comment:
All config modifications should be made in the config schema. End the config validation with coercing to int.

Reply:
@jkeljo got time to fix this?

Reply:
My mistake... I added the coerce but forgot to remove this cast. Fixed.